src/Microsoft.OData.Client/ODataRequestMessageWrapper.cs

```diff
@@ -161,6 +161,8 @@ internal static ODataRequestMessageWrapper CreateRequestMessageWrapper(BuildingR
             if (requestInfo.Timeout != 0)
             {
                 requestMessage.Timeout = requestInfo.Timeout;
+                //set readwritetimeout value as the set timeout value
+                requestMessage.ReadWriteTimeout = requestInfo.Timeout;

```

Comment:
I'm wondering if it would probably make sense to allow the user to set the 2 values separately and default to `ReadWriteTimeout = Timeout` only when `ReadWriteTimeout` is not separately defined

Reply:
resolved.